src/metabase/models/parameter_card.clj

```diff
@@ -45,12 +48,11 @@
    (delete-all-for-parameterized-object! parameterized-object-type parameterized-object-id []))
 
   ([parameterized-object-type parameterized-object-id parameter-ids-still-in-use]
-   (db/delete! ParameterCard
-               :parameterized_object_type parameterized-object-type
-               :parameterized_object_id parameterized-object-id
-               (if (empty? parameter-ids-still-in-use)
-                 {}
-                 {:where [:not-in :parameter_id parameter-ids-still-in-use]}))))
+   (let [conditions (concat [:parameterized_object_type parameterized-object-type
+                             :parameterized_object_id parameterized-object-id]
+                            (when-not (empty? parameter-ids-still-in-use)
+                              [:parameter_id [:not-in parameter-ids-still-in-use]]))]
+     (apply (partial db/delete! ParameterCard) conditions))))
```

Comment:
`(apply (partial db/delete! ParameterCard) conditions)))` is the same as `(apply db/delete! ParameterCard conditions)`

Reply:
conditions is a seq.

```
(apply db/delete! ParameterCard [:id 1 :parameter_id "ABC"])
-> 
(db/delete! Parameter Card [:id 1 :parameter_id "ABC"])


(apply (partial db/delete! ParameterCard) [:id 1 :parameter_id "ABC"])
-> 
(db/delete! Parameter Card :id 1 :parameter_id "ABC")
```

Reply:
> conditions is a seq
Exactly. 

See the definition of apply:
```
(defn apply
  "Applies fn f to the argument list formed by prepending intervening arguments to args."
  {:added "1.0"
   :static true}
  ([^clojure.lang.IFn f args]
     (. f (applyTo (seq args))))
  ([^clojure.lang.IFn f x args]
     (. f (applyTo (list* x args)))) ; <-- this is the one applied here. 
  ([^clojure.lang.IFn f x y args]
     (. f (applyTo (list* x y args))))
  ([^clojure.lang.IFn f x y z args]
     (. f (applyTo (list* x y z args))))
  ([^clojure.lang.IFn f a b c d & args]
     (. f (applyTo (cons a (cons b (cons c (cons d (spread args)))))))))
```

So from your example:
```
(apply db/delete! ParameterCard [:id 1 :parameter_id "ABC"]) 
-> 
(db/delete! Parameter Card :id 1 :parameter_id "ABC")
```

Reply:
interesting. TIL. Will change

Reply:
https://github.com/metabase/metabase/pull/27495/commits/83ef34801727f0e44b0fabf9ed02cff301c95fc5

made the change!